types/kurento-client/index.d.ts

```diff
@@ -0,0 +1,132 @@
+    gatherCandidates: (callback?: Callback<void>) => Promise<void>;
+    getConnectionState: (callback?: Callback<any>) => Promise<any>;
+    getICECandidatePairs: (callback?: Callback<any>) => Promise<any>;
+    getIceConnectionState: (callback?: Callback<IceConnection>) => Promise<IceConnection>;
+    setMaxAudioRecvBandwidth: (value: number, callback?: Callback<void>) => Promise<void>;
+    getMaxAudioRecvBandwidth: (callback?: Callback<number>) => Promise<number>;
+    setMinVideoRecvBandwidth: (value: number, callback?: Callback<void>) => Promise<void>;
+    getMinVideoRecvBandwidth: (callback?: Callback<number>) => Promise<number>;
+    setMaxVideoRecvBandwidth: (value: number, callback?: Callback<void>) => Promise<void>;
+    getMaxVideoRecvBandwidth: (callback?: Callback<number>) => Promise<number>;
+    setMinVideoSendBandwidth: (value: number, callback?: Callback<void>) => Promise<void>;
+    getMinVideoSendBandwidth: (callback?: Callback<number>) => Promise<number>;
+    setMaxVideoSendBandwidth: (value: number, callback?: Callback<void>) => Promise<void>;
+    getMaxVideoSendBandwidth: (callback?: Callback<number>) => Promise<number>;
+    setMinOutputBitrate: (value: number, callback?: Callback<void>) => Promise<void>;
+    getMinOutputBitrate: (callback?: Callback<number>) => Promise<number>;
+    setMaxOutputBitrate: (value: number, callback?: Callback<void>) => Promise<void>;
+    getMaxOutputBitrate: (callback?: Callback<number>) => Promise<number>;
+    getMediaState: (callback?: Callback<number>) => Promise<any>;
+    setStunServerAddress: (server: string, callback?: Callback<void>) => Promise<void>;
+    getStunServerAddress: (callback?: Callback<string>) => Promise<string>;
+    setStunServerPort: (port: number, callback?: Callback<void>) => Promise<void>;
+    getStunServerPort: (callback?: Callback<number>) => Promise<number>;
+    setTurnUrl: (url: string, callback?: Callback<void>) => Promise<void>;
+    getTurnUrl: (callback?: Callback<string>) => Promise<string>;
+    processOffer: (offer: string, callback?: Callback<string>) => Promise<string>;
+}
+
+declare class KurentoClient {
+    create(type: 'MediaPipeline'): Promise<MediaPipeline>;
+    create(type: 'WebRtcEndpoint'): Promise<WebRtcEndpoint>;
+    on(event: 'OnIceCandidate', callback: (event: IceCandidate) => void): void;
+    on(event: 'Error', callback: (error: Error) => void): void;
+    on(event: 'Recording' | 'Paused' | 'Stopped', callback: () => void): void;
+    getMediaobjectById(objectId: string): Promise<any>;
+    close(): void;
+}
+
+interface Kurento {
+    (ws_uri: string, options?: Options): Promise<KurentoClient>;
+    getComplexType: (complex: 'IceCandidate') => (value: any) => any;
+}
+
+declare const kurento: Kurento;
+
+// tslint:disable-next-line npm-naming
+export default kurento;
```

Comment:
Looks like the library is a CommonJS module, not an ES Module: https://unpkg.com/kurento-client@6.13.0/lib/index.js

Should be something like

```ts
declare namespace kurento {
  // exported types go here
}

declare const kurento: kurento.Kurento;

export = kurento;
```

Reply:
Yeh, the reason I structured the types like this is so an end user can import isolated interfaces from the types, so this structure:

```
import kurento, { MediaPipeline } from 'kurento-client';
...
const client = kurento(...);
...
class Something {
  private pipeline: MediaPipeline;
  ...
}
```
If this is wrong I'll make the relevant changes though. Unless there's an alternative method of providing this that I'm missing?

Thanks for the feedback!